src/vnode/src/vnodeWrite.c

```diff
@@ -77,36 +77,25 @@ int32_t vnodeProcessWrite(void *param1, int qtype, void *param2, void *item) {
   code = walWrite(pVnode->wal, pHead);
   if (code < 0) return code;
 
-  int32_t syncCode = syncForwardToPeer(pVnode->sync, pHead, item);
-  if (syncCode < 0) return syncCode;
+  code = syncForwardToPeer(pVnode->sync, pHead, item);
+  if (code < 0) return code;
 
-  code = (*vnodeProcessWriteMsgFp[pHead->msgType])(pVnode, pHead->cont, item);
+  code = (*vnodeProcessWriteMsgFp[pHead->msgType])(pVnode, pHead->cont);
   if (code < 0) return code;
 
-  return syncCode;
+  return code;
 }
 
-static int32_t vnodeProcessSubmitMsg(SVnodeObj *pVnode, void *pCont, SRspRet *pRet) {
+static int32_t vnodeProcessSubmitMsg(SVnodeObj *pVnode, void *pCont) {
   int32_t code = 0;
 
-  // save insert result into item
-
   dTrace("pVnode:%p vgId:%d, submit msg is processed", pVnode, pVnode->vgId);
   code = tsdbInsertData(pVnode->tsdb, pCont);
 
-  pRet->len = sizeof(SShellSubmitRspMsg);
-  pRet->rsp = rpcMallocCont(pRet->len);
-  SShellSubmitRspMsg *pRsp = pRet->rsp;
-
-  pRsp->code              = 0;
-  pRsp->numOfRows         = htonl(1);
-  pRsp->affectedRows      = htonl(1);
-  pRsp->numOfFailedBlocks = 0;
-
```

Comment:
these a few lines shall no t be removed, since TDengine client needs to know affected rows and other information besides the code (success or an error code). That's why we need SRspRet parameter. Create/Alter/Drop Table/DB may return some info to client even they don't return anything in current version.

Reply:
Looks like SRspRet is overwritten immediately in dnodeProcessWriteQueue() instead of returning back to the RPC module. Am I missing something here?

```
for (int32_t i = 0; i < numOfMsgs; ++i) {
      pWrite = NULL;
      taosGetQitem(pWorker->qall, &type, &item);
      if (type == TAOS_QTYPE_RPC) {
        pWrite = (SWriteMsg *)item;
        pHead = (SWalHead *)(pWrite->pCont - sizeof(SWalHead));
        pHead->msgType = pWrite->rpcMsg.msgType;
        pHead->version = 0;
        pHead->len = pWrite->contLen;
      } else {
        pHead = (SWalHead *)item;
      }

      int32_t code = vnodeProcessWrite(pVnode, type, pHead, item); <-- item gets overwritten within this loop
      if (pWrite) pWrite->rpcMsg.code = code;
    }

    walFsync(vnodeGetWal(pVnode));

    // browse all items, and process them one by one
    taosResetQitems(pWorker->qall);
    for (int32_t i = 0; i < numOfMsgs; ++i) {
      taosGetQitem(pWorker->qall, &type, &item); <-- item also gets overwritten here
      if (type == TAOS_QTYPE_RPC) {
        pWrite = (SWriteMsg *)item;
        dnodeSendRpcWriteRsp(pVnode, item, pWrite->rpcMsg.code); 
      } else {
        taosFreeQitem(item);
        vnodeRelease(pVnode);
      }
    }
```

Reply:
it wont be overwritten, since a new pWrite will be retrieved from taosGetQitem in each loop.
SRspRet is part of SWriteMsg, so the response can be passed out.

Reply:
Ah I see. I missed that part of the code. Thanks for the explanation :)